hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneClientAdapterImpl.java

```diff
@@ -643,6 +645,13 @@ public FileStatusAdapter getFileStatus(String path, URI uri,
     }
     try {
       OzoneBucket bucket = getBucket(ofsPath, false);
+
+      if (ofsPath.isSnapshotIndicator()) {
```

Comment:
It would be easier to read if written like below and exception is handle in individual function.
```
    if (ofsPath.isRoot()) {
      return getFileStatusAdapterForRoot(uri);
    } else if (ofsPath.isVolume()) {
      OzoneVolume volume = objectStore.getVolume(ofsPath.getVolumeName());
      return getFileStatusAdapterForVolume(volume, uri);
    } else if (ofsPath.isSnapshotIndicator()) {
      OzoneVolume volume = objectStore.getVolume(ofsPath.getVolumeName());
      return getFileStatusAdapterWithSnapshotIndicator(
          volume, bucket, uri);
    } else {
      OzoneFileStatus status = bucket.getFileStatus(key);
      return toFileStatusAdapter(status, userName, uri, qualifiedPath,
          ofsPath.getNonKeyPath());
    }
```

Reply:
I think it's worth keeping the try catch, in order to have a more specific error message in case of failure. It's gonna look like so 

```suggestion
      if (ofsPath.isRoot()) {
        ...
      } else if (ofsPath.isVolume()) {
        ...
      } else {
        try {
          if (ofsPath.isSnapshotIndicator()) {
            ...
          } else {
            ...
          }
        } catch () {
          ...
        } 
      }
```

Reply:
Don't have strong point to move try-catch to individual function. One thing it will improve is nesting. Right now we have too much nesting.

Reply:
@hemantk-12 I get that you want to get rid of nesting. I don't understand what you mean to do with the try-catch. Do you want me to wrap all the `if - else if` block with the try-catch or just wrap whatever line might throw an exception? Or do you want me to remove the try-catch altogether?

Reply:
ooh, now I get it. You are saying to move 
```
        try {
          if (ofsPath.isSnapshotIndicator()) {
            ...
          } else {
            ...
          }
        } catch () {
          ...
        } 
```
to a new function.